### ISSUE TITLE
Apache/PHP connection not working in 2.1.2 on Windows 7

### ISSUE THREAD

**Issue by itsgreggmarshall** [geerlingguy/drupal-vm#387]
I have a feeling there is some interaction with the configuration using 2.1.2 (required to have get past the ansible_local issue #372 on Windows 7.  While the build completes with no errors reported

     drupalvm: localhost                  : ok=198  changed=99   unreachable=0 failed=0

When I visit drupalvm.dev on my local machine I get the PHP listing instead of executed.  SSHing into the server shows index.php is executable.  And PHP -v gives 

     PHP 5.6.17-3+deb.sury.org~trusty+1 (cli)
     Copyright (c) 1997-2015 The PHP Group
     Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
         with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies
         with Xdebug v2.2.5, Copyright (c) 2002-2014, by Derick Rethans

Not sure what to look for, right now drupal-vm is pretty much a black box.

**Comment by geerlingguy** [comment]
@itsgreggmarshall - Try setting up your virtualhost like the updated example and see if that works:

https://github.com/geerlingguy/drupal-vm/blob/master/example.config.yml#L79-L83

**Comment by claudfernandes** [comment]
This is what I did to work

```
apache_vhosts:
  - servername: "{{ drupal_domain }}"
    documentroot: "{{ drupal_core_path }}"
    extra_parameters: |
      ProxyPassMatch ^/(.*\.php(/.*)?)$ "unix:/var/run/php5-fpm.sock|fcgi://127.0.0.1:9000{{ drupal_core_path }}"

  - servername: "adminer.drupalvm.dev"
    documentroot: "/opt/adminer"
    extra_parameters: |
      ProxyPassMatch ^/(.*\.php(/.*)?)$ "unix:/var/run/php5-fpm.sock|fcgi://127.0.0.1:9000/opt/adminer"

  - servername: "xhprof.drupalvm.dev"
    documentroot: "/usr/share/php/xhprof_html"
    extra_parameters: |
      ProxyPassMatch ^/(.*\.php(/.*)?)$ "unix:/var/run/php5-fpm.sock|fcgi://127.0.0.1:9000/usr/share/php/xhprof_html"


  - servername: "pimpmylog.drupalvm.dev"
    documentroot: "/usr/share/php/pimpmylog"
    extra_parameters: |
      ProxyPassMatch ^/(.*\.php(/.*)?)$ "unix:/var/run/php5-fpm.sock|fcgi://127.0.0.1:9000/usr/share/php/pimpmylog"
```

**Comment by geerlingguy** [comment]
@claudfernandes - Ah, you're right, we'll need to add the extra_parameters to some of the other subdomains as well.

**Comment by itsgreggmarshall** [comment]
I thought I had tried that, but I put in all 4 from @claudfernandes and get an error when apache is restarted:

     ==> drupalvm: RUNNING HANDLER [geerlingguy.apache : restart apache] **************************
     ==> drupalvm: fatal: [localhost]: FAILED! => {"changed": false, "failed": true,
     "msg": "Output of config test was:\nAH00526: Syntax error on line 13 of /etc/apache2/sites-   enabled/vhosts.conf:\nInvalid command 'ProxyPassMatch', perhaps misspelled or defined by a module not included in the server configuration\nAction 'configtest' failed.\nThe Apache error log may have more information.\n"}

Could one of my previous tries with 2.2.0 and head have pulled down a later box that isn't compatible with the 2.1.2 version?  If so, is there a way to specify the box version to use?

**Comment by geerlingguy** [comment]
The boxes themselves are purposely generic (e.g. `geerlingguy/ubuntu1404` is simply an Ubuntu 14.04 minimal server x64 install), so the box shouldn't be an issue.

However, if you did the first `vagrant up` with 2.1.2 then switched to 2.2.0 and did a provision, then switched back, things will be broken (or vice-versa). When switching versions of Drupal VM (especially if going backwards), it's best to do a full `vagrant destroy` then `vagrant up`.

**Comment by geerlingguy** [comment]
One other thing—if you're using 2.1.2, then you should add `libapache2-mod-php5` to the 'extra packages' list in config.yml—otherwise mod_php isn't installed anymore, according to the php role defaults in it's latest version.

Sorry about all the mess—I didn't plan on Vagrant having an annoying bug with it's ansible_local provisioner when I was planning on the Apache mod_php to php-fpm transition :(

**Comment by itsgreggmarshall** [comment]
No problem, I was assuming the transition would be a challenge.  I put the libapache2-mod-php5 under  unzip.

Now I get a new error:
    ==> drupalvm: Setting hostname...
    An error occurred in the underlying SSH library that Vagrant uses.
    The error message is shown below. In many cases, errors from this
    library are caused by ssh-agent issues. Try disabling your SSH
    agent or removing some keys and try again.

    If the problem persists, please report a bug to the net-ssh project.

    timeout during server version negotiating

**Comment by geerlingguy** [comment]
@itsgreggmarshall - Can you do a `vagrant destroy -f` and start over? Otherwise a `vagrant reload --provision`. Sounds like Vagrant and VirtualBox aren't happy with each other right now :P

**Comment by geerlingguy** [comment]
btw, I've dusted off my Win10 laptop and will be testing some things today to see what other workarounds I can come up with.

**Comment by itsgreggmarshall** [comment]
So far so go using destroy -f, off to a meeting while it runs.

BTW this is Windows 7, my Windows 10 is happy (didn't upgrade it).

**Comment by itsgreggmarshall** [comment]
I spoke too soon, it failed at the same place, just took longer to fail so I was in the meeting.

I tried a couple of other options (deleting all trace of the VM, rebooting Windows, etc) then uninstalled VirtualBox and Vagrant and reinstalled them (I let Vagrant tell me what version of VirtualBox to use).  Switching to VirtualBox 5.0.10 from 5.0.14 solved the SSH key problem, but gave me a new one.

    ==> drupalvm: RUNNING HANDLER [geerlingguy.apache : restart apache] **************************
     ==> drupalvm: fatal: [localhost]: FAILED! => {"changed": false, "failed": true,
     "msg": "Output of config test was:\n[Thu Jan 21 12:49:23.727570 2016] [:crit] [pid 31974:tid  140629869143936] Apache is running a threaded MPM, but your PHP Module is not compiled to be threadsafe.  You need to recompile PHP.\nAH00013: Pre-configuration failed\nAction 'configtest' failed.\nThe Apache error log may have more information.\n"}

I presume the issue is coming from adding libapache2-mod-php5 as an extra_package?

**Comment by afbrogneaux** [comment]
@itsgreggmarshall Same error here (directly in the VM shell):
```
vagrant@drupalvm:~$ apachectl configtest
[Thu Jan 21 12:39:19.549976 2016] [:crit] [pid 29678:tid 139793749469056] Apache is running a threaded MPM, but your PHP Module is not compiled to be threadsafe.  You need to recompile PHP.
AH00013: Pre-configuration failed
Action 'configtest' failed.
The Apache error log may have more information.
```

**Comment by itsgreggmarshall** [comment]
Pretty much each time I nuke the VM (vagrant destroy -f) and start over I'm getting a new/different issue.

The latest iteration seems to not like the ProxyPassMatch:

    ==> drupalvm: RUNNING HANDLER [geerlingguy.apache : restart apache] **************************
    ==> drupalvm: fatal: [localhost]: FAILED! => {"changed": false, "failed": true,
    "msg": "Output of config test was:\nAH00526: Syntax error on line 13 of /etc/apache2/sites-enabled/vhosts.conf:\nInvalid command 'ProxyPassMatch', perhaps misspelled or defined by a module not included in the server configuration\nAction 'configtest' failed.\nThe Apache error log may have more information.\n"}

Actually that might make sense since I commented out the libapache2-mod-php5 line to see if that was the previous error I saw.

So maybe the next step is to figure out how to get that package to load the threaded version?

**Comment by afbrogneaux** [comment]
BTW, the same issue happens with Windows 10.

**Comment by itsgreggmarshall** [comment]
I have a feeling even 2.1.2 is using ansible 2, looking through the messages that are generated during a provision, I saw this:

    Downloading ansible-2.0.0.2.tar.gz (1.5MB)

    ==> drupalvm: Successfully installed ansible-2.0.0.2

**Comment by geerlingguy** [comment]
@itsgreggmarshall - 2.1.2 is using Ansible 2.x within windows (it will still install the latest version available), but it doesn't use the `ansible_local` provisioner (it uses the now-deprecated JJG-Ansible-Windows script instead), so if there's an issue with the provisioning post-Ansible installation, that means Ansible is at least working.

I'm doing some more testing here, now that I have my Windows laptop out (Win 10 i5 with all the latest versions of everything), and will see what I can find with both 2.1.2 and 2.2.0.

**Comment by afbrogneaux** [comment]
I'm not sure to understand how to use this commit. Should we still use 2.1.2 but with the updated config.yml file of 2.2.1?

**Comment by itsgreggmarshall** [comment]
You can use 2.1.2 but some of use ran into an issue with PHP and Apache that was resolved in issue #361.  I found with the patches to vagrant in #361, I could use 2.2.1.
